identity/src/main/java/com/android/identity/document/Document.kt

```diff
@@ -242,9 +242,6 @@ class Document private constructor(
     fun addCredential(
         credential: Credential,
     ) {
-        check(_pendingCredentials.none { it.identifier == credential.identifier }) {
-            "There is already a pending credential with the same identifier as the given credential"
-        }
```

Comment:
We could keep this check if the `identifier` uses the `createUniqueAlias` function you added, which would be helpful to prevent users accidentally adding the same cred to the doc twice

Reply:
The problem is that we don't assign an identifier until we actually assign the Credential to the Document.

Reply:
Ideally the credential would be added in the constructor so the app won't have to... it's not clear to be that this isn't impossible. Let's chat!